integration-tests/scripts/baseline/main.py

```diff
@@ -46,22 +46,78 @@
 # in the empty document to be explicit.
 EMPTY_BASELINE_STRUCTURE = []
 
+# In which format we expect to find dates in the baseline files
+DATE_FORMAT = "%Y-%m-%d %H:%M:%S"
+
+
+def object_parser_hook(data):
+    result = data
+
+    for f in (datetime_parser, memory_parser):
+        result = f(result)
+
+    return result
+
+
+def memory_parser(data):
+    result = {}
+    multipliers = {
+        "GiB": 1024,
+        "MiB": 1,
+        "KiB": 1 / 1024,
+    }
+
+    for k, v in data.items():
+        if k in ('Mem') and isinstance(v, str) and len(v) > 3:
+            suffix = v[-3:]
+            multiplier = multipliers.get(suffix, 0)
+
+            if multiplier == 0:
+                result[k] = 0
+            else:
+                result[k] = float(v[:-3]) * multiplier
+        else:
+            result[k] = v
+
+    return result
+
+
+def datetime_parser(data):
+    result = {}
+
+    for k, v in data.items():
+        if k in ('Timestamp', 'LoadStartTs', 'LoadStopTs'):
```

Comment:
Do we want to validate the type of `v` here, in the same way as we do for memory?

Reply:
I think the act of parsing the datetime serves as validation here. Or do you think something more rigorous is needed?

Reply:
No, that's a fair point, it can be left as-is